static/js/import_wizard/components/value_map_section.tsx

```diff
@@ -0,0 +1,236 @@
+            </div>
+            {mappingInputOrderedList.map((id) => {
+              return (
+                <div className="value-map-input-entry" key={id}>
+                  <Input
+                    className="value-map-input"
+                    type="text"
+                    onChange={(e) =>
+                      onMappingInputChanged(id, e.target.value, true)
+                    }
+                    value={mappingInput[id].originalVal}
+                  />
+                  <Input
+                    className="value-map-input"
+                    type="text"
+                    onChange={(e) =>
+                      onMappingInputChanged(id, e.target.value, false)
+                    }
+                    value={mappingInput[id].newVal}
+                  />
+                  <i
+                    className="material-icons-outlined"
+                    onClick={() => {
+                      setMappingInput((prev) => {
+                        const newMappingInput = _.cloneDeep(prev);
+                        delete newMappingInput[id];
+                        return newMappingInput;
+                      });
+                    }}
+                  >
+                    cancel
+                  </i>
+                </div>
+              );
+            })}
+            <div
+              className="value-map-add-entry"
+              onClick={() => {
+                const nextId = _.isEmpty(mappingInputOrderedList)
+                  ? 0
+                  : Number(
+                      mappingInputOrderedList[
+                        mappingInputOrderedList.length - 1
+                      ]
+                    ) + 1;
+                setMappingInput((prev) => {
+                  return { ...prev, [nextId]: EMPTY_VALUE_MAP_INPUT };
+                });
+              }}
+            >
+              <i className="material-icons-outlined">add_circle</i>
+              Add entry
+            </div>
+          </div>
+        )}
+      </div>
+      {showConfirmationButton && (
+        <div className="confirmation-button">
+          <Button onClick={onConfirmationClicked}>Next</Button>
+        </div>
+      )}
+    </>
+  );
+
+  function onMappingInputChanged(
+    inputId: string,
+    newInputVal: string,
+    isOriginalVal: boolean
+  ): void {
+    setMappingInput((prev) => {
+      const newMappingInput = _.cloneDeep(prev);
+      newMappingInput[inputId] = {
+        originalVal: isOriginalVal ? newInputVal : prev[inputId].originalVal,
+        newVal: isOriginalVal ? prev[inputId].newVal : newInputVal,
+      };
+      return newMappingInput;
+    });
+  }
+
+  function onConfirmationClicked(): void {
+    const valueMap = {};
+    if (showFileUpload) {
+      const reader = new FileReader();
+      reader.onload = (e) => {
+        const rows = Papa.parse(e.target.result as string).data;
+        for (const row of rows) {
+          if (row.length !== 2) {
+            continue;
```

Comment:
if it's not a blank line, should we display an error? e.g. Skipping ___ because there was no value specified.

Reply:
actually, if there's only one value, should we map that value to blank in the cleaned csv?

Reply:
This would still need to be specified as 
`bad_value,`
with the trailing comma, so i would imagine that papaparse would return that as 2 values

Reply:
+1 to having the capability of mapping bad_value to empty

Reply:
also, might be worth having unit-test for this parsing?

Reply:
done